editor/editor_audio_buses.cpp

```diff
@@ -665,56 +697,84 @@ void EditorAudioBus::drop_data_fw(const Point2 &p_point, const Variant &p_data,
 			paste_at--;
 		}
-	} else {
-		paste_at = -1;
-	}
 
-	bool enabled = AudioServer::get_singleton()->is_bus_effect_enabled(bus, effect);
+		bool enabled = AudioServer::get_singleton()->is_bus_effect_enabled(bus, effect);
 
-	UndoRedo *ur = EditorNode::get_undo_redo();
-	ur->create_action(TTR("Move Bus Effect"));
-	ur->add_do_method(AudioServer::get_singleton(), "remove_bus_effect", bus, effect);
-	ur->add_do_method(AudioServer::get_singleton(), "add_bus_effect", get_index(), AudioServer::get_singleton()->get_bus_effect(bus, effect), paste_at);
+		UndoRedo *ur = EditorNode::get_undo_redo();
+		ur->create_action(TTR("Move Bus Effect"));
+		ur->add_do_method(AudioServer::get_singleton(), "remove_bus_effect", bus, effect);
+		ur->add_do_method(AudioServer::get_singleton(), "add_bus_effect", get_index(), AudioServer::get_singleton()->get_bus_effect(bus, effect), paste_at);
+
+		if (paste_at == -1) {
+			paste_at = AudioServer::get_singleton()->get_bus_effect_count(get_index());
+			if (bus == get_index()) {
+				paste_at--;
+			}
+		}
+		if (!enabled) {
+			ur->add_do_method(AudioServer::get_singleton(), "set_bus_effect_enabled", get_index(), paste_at, false);
+		}
 
-	if (paste_at == -1) {
-		paste_at = AudioServer::get_singleton()->get_bus_effect_count(get_index());
-		if (bus == get_index()) {
-			paste_at--;
+		ur->add_undo_method(AudioServer::get_singleton(), "remove_bus_effect", get_index(), paste_at);
+		ur->add_undo_method(AudioServer::get_singleton(), "add_bus_effect", bus, AudioServer::get_singleton()->get_bus_effect(bus, effect), effect);
+		if (!enabled) {
+			ur->add_undo_method(AudioServer::get_singleton(), "set_bus_effect_enabled", bus, effect, false);
 		}
-	}
-	if (!enabled) {
-		ur->add_do_method(AudioServer::get_singleton(), "set_bus_effect_enabled", get_index(), paste_at, false);
-	}
 
-	ur->add_undo_method(AudioServer::get_singleton(), "remove_bus_effect", get_index(), paste_at);
-	ur->add_undo_method(AudioServer::get_singleton(), "add_bus_effect", bus, AudioServer::get_singleton()->get_bus_effect(bus, effect), effect);
-	if (!enabled) {
-		ur->add_undo_method(AudioServer::get_singleton(), "set_bus_effect_enabled", bus, effect, false);
+		ur->add_do_method(buses, "_update_bus", get_index());
+		ur->add_undo_method(buses, "_update_bus", get_index());
+		if (get_index() != bus) {
+			ur->add_do_method(buses, "_update_bus", bus);
+			ur->add_undo_method(buses, "_update_bus", bus);
+		}
+		ur->commit_action();
 	}
 
-	ur->add_do_method(buses, "_update_bus", get_index());
-	ur->add_undo_method(buses, "_update_bus", get_index());
-	if (get_index() != bus) {
-		ur->add_do_method(buses, "_update_bus", bus);
-		ur->add_undo_method(buses, "_update_bus", bus);
+	if (String(d["type"]) == "files") {
+		Vector<String> files = d["files"];
+		for (const String &E : files) {
+			String ftype = EditorFileSystem::get_singleton()->get_file_type(E);
+
+			if (!ClassDB::is_parent_class(ftype, "AudioEffect")) {
+				return;
+			}
+		}
```

Comment:
This basically repeats the code in `can_drop_data()`. Either you can move it to a new method (I've seen a pattern where `drop_data()` calls `is_drop_valid()` at the beginning), but it would be better if you just moved the check to the loop below. If invalid file gets past `can_drop_data()` (which is IMO impossible unless you call the method manually), it will be simply ignored.